bofire/data_models/strategies/predictives/botorch.py

```diff
@@ -120,28 +126,30 @@ def _generate_surrogate_specs(
         Returns:
             List[ModelSpec]: List of model specification classes
         """
-        existing_keys = (
-            surrogate_specs.outputs.get_keys() if surrogate_specs is not None else []
-        )
+        existing_keys = surrogate_specs.outputs.get_keys()
         non_exisiting_keys = list(set(domain.outputs.get_keys()) - set(existing_keys))
-        _surrogate_specs = (
-            surrogate_specs.surrogates if surrogate_specs is not None else []
-        )
+        _surrogate_specs = surrogate_specs.surrogates
         for output_feature in non_exisiting_keys:
             if len(domain.inputs.get(CategoricalInput, exact=True)):
                 _surrogate_specs.append(
                     MixedSingleTaskGPSurrogate(
                         inputs=domain.inputs,
-                        outputs=Outputs(features=[domain.outputs.get_by_key(output_feature)]),  # type: ignore
+                        outputs=Outputs(
+                            features=[domain.outputs.get_by_key(output_feature)]  # type: ignore
+                        ),  # type: ignore
                     )
                 )
             else:
                 _surrogate_specs.append(
                     SingleTaskGPSurrogate(
                         inputs=domain.inputs,
-                        outputs=Outputs(features=[domain.outputs.get_by_key(output_feature)]),  # type: ignore
+                        outputs=Outputs(
+                            features=[
+                                domain.outputs.get_by_key(output_feature)  # type:ignore
+                            ]
+                        ),  # type: ignore
```

Comment:
When I understood this correctly, `# type: ignore` correctly, it operates on the line level. We could at some point activate `reportUnnecessaryTypeIgnoreComment` in Pyright's options.
See https://github.com/microsoft/pyright/blob/main/docs/comments.md#line-level-diagnostic-suppression